packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectInternal.kt

```diff
@@ -41,6 +48,89 @@ internal interface RealmObjectInternal : RealmObject, RealmStateHolder, io.realm
     override fun realmState(): RealmState {
         return `$realm$Owner` ?: UnmanagedState
     }
+
+    override fun freeze(frozenRealm: RealmReference): RealmObjectInternal {
+        @Suppress("UNCHECKED_CAST")
+        val type: KClass<RealmObjectInternal> = this::class as KClass<RealmObjectInternal>
+        val mediator = `$realm$Mediator`!!
+        val managedModel = mediator.createInstanceOf(type)
+        return managedModel.manage(
+            frozenRealm!!,
+            mediator,
+            type,
+            RealmInterop.realm_object_freeze(
+                `$realm$ObjectPointer`!!,
+                frozenRealm.dbPointer
+            )
+        )
+    }
+
+    override fun thaw(liveRealm: RealmReference): Observable<RealmObjectInternal>? {
+        @Suppress("UNCHECKED_CAST")
+        val type: KClass<*> = this::class
+        val mediator = `$realm$Mediator`!!
+        val managedModel = mediator.createInstanceOf(type)
+        val dbPointer = liveRealm.dbPointer
+        @Suppress("TooGenericExceptionCaught")
+        try {
+            val realmObjectThaw: NativePointer? =
+                RealmInterop.realm_object_thaw(`$realm$ObjectPointer`!!, dbPointer)
+            val realmObjectThaw1: NativePointer = realmObjectThaw!!
```

Comment:
Why not just `!!` in `realmObjectThaw`?

Reply:
I guess according to https://github.com/realm/realm-kotlin/pull/440 the `thaw` will no longer throw but rather return a `null` for invalid/deleted object

Reply:
Looks like I just introduced an intermediate variable for some debugging/experiments - Removed it again. #440 will update API to match that the underlying C-API will return null on delete objects.